packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -53,32 +62,45 @@ export function objectToAST(
     );
 }
 
-export function containsDynamicChildren(children: ChildNode[]): boolean {
-    return children.some((child) => {
-        if (isForBlock(child) || isIf(child)) {
-            return containsDynamicChildren(child.children);
+export function containsDynamicChildren(parent: ParentNode): boolean {
+    const hasDynamicChildren = parent.children.some((child) => {
+        // The child in the children array will only ever contain an IfBlock.
+        // ElseIfBlock and ElseBlock are chained together starting from the IfBlock.
+        if (isForBlock(child) || isIf(child) || isIfBlock(child)) {
```

Comment:
It's not automatically considered dynamic. Here, we're just adding IfBlock to the list of nodes where we need to continue traversing. There's no special treatment of if/elseif/else as dynamic.

Reply:
I'm still not convinced that the `ifBlock` should be included in this `containsDynamicChildren`. because the `if` will always generate 1 vnode (or `null`).

I fail to see how not having the `ifBlock` here will affect the routine which is [using](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/template-compiler/src/codegen/index.ts#L161) it. if the parent node has only 1 element, it won't be the `ifBlock` (shouldFlatten returns false).